### PR TITLE
Staking v3 improvements, part 7

### PR DESCRIPTION
**Pull request by bobo-k2**
**Pull Request Summary**

- warning about losing bonus rewards on unstaking page
- enable dApp editing from owner page
- registration page styling fix (light theme text inputs color)
- fixed navigation issues from owner and dApp pages.
- new data fields: TVL %, TVV %, bonus eligible tokens, bonus pool
- leaderboard fix to not remove a dApp after dApp claiming rewards

image
image

**Check list**

- [ ] contains breaking changes
- [ ] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

### CODE REVIEW

**Comment by Kahonnohak** [comment]
I don't find the TVL and TVV here?

Screenshot 2024-01-18 at 22 33 26

**Comment by bobo-k2** [comment]
> @bobo-k2 could you add Bonus APR and Basic Rewards on Voting page top please (the were missing but we should have them here)

I merged main to this branch. APR is there
image

**Comment by bobo-k2** [comment]
> I don't find the TVL and TVV here?

Check now :)
image

**Comment by Kahonnohak** [comment]
Gorgeous thanks, I see some styling bugs and wording problems but I'll summarise that later.

For this PR, Registration still has a problem - I think because we removed the licence. could you take a look?

An unexpected error occured during dApp registration. Please screenshot this message and send to the Astar team. "Value for argument \"data\" is not a valid Firestore document. Cannot use \"undefined\" as a Firestore value (found in field \"license\"). If you want to ignore undefined values, enable `ignoreUndefinedProperties`."

**Comment by Kahonnohak** [comment]
Also this should be locked amount IMO. 
Screenshot 2024-01-19 at 10 41 08

**Comment by bobo-k2** [comment]
> For this PR, Registration still has a problem - I think because we removed the licence. could you take a look?

It should be fine now. Please try again

**Comment by bobo-k2** [comment]
> Also this should be locked amount IMO. 

Makes sense. Updated
image

**Comment by impelcrypto** [comment]
@bobo-k2 Not sure if this is related to this PR, but I can't find any dApps in Voting page

**Comment by impelcrypto** [comment]
I got some errors 
image

**Comment by bobo-k2** [comment]
> I got some errors

@Kahonnohak @impelcrypto errors are fixed.
Actually there is still one (nothing serious) which is coming from v2 code because dapp staking v2 pallet has been removed from Shibuya.
image

File: src/v2/repositories/implementations/DappStakingRepository.ts [hunk 1]
```diff
@@ -252,6 +252,10 @@ export class DappStakingRepository implements IDappStakingRepository {
     if (!DappStakingRepository.isEraSubscribed) {
       DappStakingRepository.isEraSubscribed = true;
       const api = await this.api.getApi();
+      if (!api.query.hasOwnProperty('dappsStaking')) {
```
Comment (impelcrypto) [inline src/v2/repositories/implementations/DappStakingRepository.ts:255]: Let's not use `hasOwnProperty` method from now on ([ref](https://ourcodeworld.com/articles/read/1425/how-to-fix-eslint-error-do-not-access-objectprototype-method-hasownproperty-from-target-object-no-prototype-builtins)). You can use `Object.prototype.hasOwnProperty.call` or `hasProperty` function from Astar-js ([code](https://github.com/AstarNetwork/astar.js/blob/8484da66fe096acd8e2aa0e0ab3b9c4feb0d018c/packages/sdk-core/src/modules/common/index.ts#L10)) 

btw, I think you can just change this line to below code should be enough.
``` 
 if (checkIsDappStakingV3(api)) {
...
}
```

**Review by impelcrypto — CHANGES_REQUESTED** [review]

File: src/v2/repositories/implementations/DappStakingRepository.ts (hunk 1 above)
Comment (bobo-k2) [reply src/v2/repositories/implementations/DappStakingRepository.ts]: Fixed in https://github.com/AstarNetwork/astar-apps/pull/1137/commits/17d3fe839290c98b4fc0b669318d499f83a0d8ec

**Review by impelcrypto — APPROVED** [review]

**Review by ayumitk — APPROVED** [review]

**Review by Kahonnohak — APPROVED** [review]
Great work!

**Pull request merged**